src/server/transaction.h

```diff
@@ -443,6 +441,25 @@ class Transaction {
     EventCount ec_{};
   };
 
+  // "Single claim - single modification" barrier. Multiple threads might try to claim it, only one
+  // will succeed and will be allowed to modify the guarded object until it closes the barrier.
+  // A closed barrier can't be claimed again or re-used in any way.
+  class SingleClaimBarrier {
```

Comment:
totally subjective - maybe to call it `BatonBarrier` ?
i.e.

![Uploading image.png…]()

Reply:
Indeed, I only know of folly::Baton and it's somewhat different, but in some way it makes sense and is a more entertaining name 😆